R/step_adjust_latency.R

```diff
@@ -0,0 +1,274 @@
+#'   left by leading/lagging (defaults to NA).
+#' @param skip A logical. Should the step be skipped when the
+#'  recipe is baked by [bake()]? While all operations are baked
+#'  when [prep()] is run, some operations may not be able to be
+#'  conducted on new data (e.g. processing the outcome variable(s)).
+#'  Care should be taken when using `skip = TRUE` as it may affect
+#'  the computations for subsequent operations.
+#' @param id A unique identifier for the step
+#' @template step-return
+#'
+#' @details The step assumes that the pipeline has already applied either
+#'   `step_epi_ahead` or `step_epi_lag` depending on the value of
+#'   `"method"`, and that `step_epi_naomit` has NOT been run.
+#'
+#' The `prefix` and `id` arguments are unchangeable to ensure that the code runs
+#' properly and to avoid inconsistency with naming. For `step_epi_ahead`, they
+#' are always set to `"ahead_"` and `"epi_ahead"` respectively, while for
+#' `step_epi_lag`, they are set to `"lag_"` and `"epi_lag`, respectively.
+#'
+#' @family row operation steps
+#' @rdname step_adjust_latency
+#' @export
+#' @examples
+#' r <- epi_recipe(case_death_rate_subset) %>%
+#'   step_epi_ahead(death_rate, ahead = 7) %>%
+#'   #   step_adjust_latency(method = "extend_ahead") %>%
+#'   step_epi_lag(death_rate, lag = c(0, 7, 14))
+#' r
+step_adjust_latency <-
+  function(recipe,
+           ...,
+           role = NA,
+           trained = FALSE,
+           method = c(
+             "extend_ahead",
+             "locf",
+             "extend_lags"
+           ),
+           fixed_latency = NULL,
+           fixed_asof = NULL,
+           default = NA,
+           skip = FALSE,
+           columns = NULL,
+           id = recipes::rand_id("epi_lag")) {
+    arg_is_chr_scalar(id, method)
+    if (!is_epi_recipe(recipe)) {
+      cli::cli_abort("This recipe step can only operate on an `epi_recipe`.")
+    }
+    if (!is.null(columns)) {
+      cli::cli_abort(c("The `columns` argument must be `NULL`.",
+        i = "Use `tidyselect` methods to choose columns to lag."
+      ))
+    }
+
+    method <- rlang::arg_match(method)
+    terms_used <- recipes_eval_select(enquos(...), recipe$template, recipe$term_info)
+    if (length(terms_used) == 0) {
+      terms_used <- recipe$term_info %>%
+        filter(role == "raw") %>%
+        pull(variable)
+    }
+    if (method == "extend_ahead") {
+      rel_step_type <- "step_epi_ahead"
+      shift_name <- "ahead"
+    } else if (method == "extend_lags") {
+      rel_step_type <- "step_epi_lag"
+      shift_name <- "lag"
+    }
+    relevant_shifts <- construct_shift_tibble(terms_used, recipe, rel_step_type, shift_name)
+
+    if (!any(map_lgl(
+      recipe$steps,
+      function(recipe_step) inherits(recipe_step, rel_step_type)
+    ))) {
+      cli:cli_abort("there is no `{rel_step_type}` defined before this. for the method `extend_{shift_name}` of `step_adjust_latency`, at least one {shift_name} must be previously defined.")
```

Comment:
```suggestion
      cli:cli_abort(c("There is no `{rel_step_type}` defined before this.",
         "For the method `extend_{shift_name}` of `step_adjust_latency()`,",
         "at least one {shift_name} must be previously defined."
        ))
```

Reply:
unfortunately this renders as 3 separate lines. Going to need to insert a `glue`. Unless you were suggesting the linebreak?